content/docs/user-guide/experiment-management/experiments-overview.md

```diff
@@ -1,10 +1,29 @@
 # DVC Experiments Overview
 
-DVC Experiments are captures automatically by DVC when you [run them].
+DVC Experiments are captures automatically by DVC when you [run them]. Each
+experiment creates and tracks a project variation based on the changes in your
+<abbr>workspace</abbr>. Experiments preserve the latest commit in the current
+branch (Git `HEAD`) as their parent or _baseline_.
+
+<details>
+
+### ⚙️ How does DVC track experiments?
+
+Experiments are custom [Git references] (found in `.git/refs/exps`) with a
+single commit based on `HEAD` (not checked out by DVC). Note that these commits
+are not pushed to Git remotes by default (see `dvc exp push`).
+
+</details>
+
+Experiments will have an auto-generated ID like `exp-bfe64` by default. A custom
+name can be given instead (using the `--name`/`-n` option of `dvc exp run`).
```

Comment:
Also (@iesahin), [Running Experiments](https://dvc.org/doc/user-guide/experiment-management/running-experiments) doesn't cover `--name` so I had to mention the command directly from here. It'd be ideal if there's a section there to link from here instead...

Reply:
Added a check box about this to https://github.com/iterative/dvc.org/issues/2768.